templates/mainapp/layout.html

```diff
@@ -125,7 +125,17 @@ <h4 class="modal-title" id="booktitle">Basic Modal</h4>
                             <p id="bookauthors">
                             </p>
 
-                            <div id="bookrating"></div>
+                            {% if user.is_authenticated %}
+                            <div class='rating-block-modal' id='rating-block-modal'
+                            onclick='rateBookModal(this)' onmouseout='finishRatesModal(this)'>
```

Comment:
@sk1122 Instead of making a new global variable, why don't you set a new attribute on this element as `data-book-id` when the modal is modified in the function at Line 383 then you won't have to use the new variable in the function at  Line 318, Does that make sense?

Reply:
I tried that but how to access the id,

`explore.html` template is being rendered by function and `layout.html` `extends` in explore.html, So we cannot send variables directly to `layout.html` so we cannot set `data-book-id` in layout.html

Reply:
Set the attribute using setattr or a similar function in js on line 383 when the modal is modified